### ISSUE TITLE
Error after upgrading to v0.21.0 (from v0.20.0), Reconcilier error, getting drift for node, providerRef must be defined for Drift

### ISSUE THREAD

**Issue by tiago156** [aws/karpenter#3120]
### Version

__Karpenter Version:__ v0.21.0

__Kubernetes Version:__ v1.24.7-eks-fb459a0

### Expected Behavior

Upgrade from version v0.20.0 to v0.21.0 and everything keep working. I searched for migration notes from v0.20.0 to v0.21.0 and didn't found any, thats why I'm opening this bug report.

### Actual Behavior

Getting error message every second on karpenter pod:

`2022-12-28T18:50:21.135Z	ERROR	controller	Reconciler error	{"commit": "0c8536a-dirty", "controller": "drift", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-10-0-61-28.ec2.internal"}, "namespace": "", "name": "ip-10-0-61-28.ec2.internal", "reconcileID": "f8cf97ba-6a9e-4e79-8377-693b26811ad9", "error": "getting drift for node, providerRef must be defined for Drift"}`

Changing configmap feature key `featureGates.driftEnabled` to "true", doesn't change this behavior (not sure if it should, but tried anyway), Reconciler error keep spamming the pod log.

### Steps to Reproduce the Problem

Upgrade from karpenter v0.20.0 to v0.21.0 using this provisioner:

```
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  annotations:
  generation: 4
  name: amd64
spec:
  consolidation:
    enabled: true
  limits:
    resources:
      cpu: "100"
  provider:
    blockDeviceMappings:
    - deviceName: /dev/xvda
      ebs:
        deleteOnTermination: true
        encrypted: true
        volumeSize: 50Gi
        volumeType: gp3
    securityGroupSelector:
      aws-ids: sg-xxxxxxxxxxxxxxx
    subnetSelector:
      aws-ids: subnet-yyyyyyyyyyyyyyyy,subnet-xxxxxxxxxxxxxxx
    tags:
      karpenter.sh/discovery: ABC
  requirements:
  - key: karpenter.sh/capacity-type
    operator: In
    values:
    - spot
  - key: karpenter.k8s.aws/instance-family
    operator: In
    values:
    - r5
    - r5d
    - r5b
    - r5dn
    - r6a
    - r6i
    - r6id
    - r6in
    - r6idn
  - key: kubernetes.io/arch
    operator: In
    values:
    - amd64
  - key: karpenter.k8s.aws/instance-size
    operator: In
    values:
    - medium
    - large
    - xlarge
  - key: kubernetes.io/os
    operator: In
    values:
    - linux
  weight: 1
```

### Resource Specs and Logs

Pod log: [karpenter.log](https://github.com/aws/karpenter/files/10315951/karpenter.log)
Nodes (kubectl get nodes -o yaml): [nodes.txt](https://github.com/aws/karpenter/files/10316274/nodes.txt)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
Hey @tiago156, this was intentional as part of Drift. We've deprecated the in-line provider spec, so you should use a providerRef with an AWSNodeTemplate.

**Comment by ellistarn** [comment]
Thanks for this report! We're working on a patch release to resolve this ASAP.

**Comment by ellistarn** [comment]
Fixed in v0.21.1
